test/extensions/filters/http/cache/cache_filter_test.cc

```diff
@@ -172,407 +298,290 @@ TEST_F(CacheFilterTest, UncacheableResponse) {
-  }
-  waitBeforeSecondRequest();
-  {
-    // Create filter for request 2
-    CacheFilterSharedPtr filter = makeFilter(simple_cache_);
-
-    // Make request require validation
-    request_headers_.setReferenceKey(Http::CustomHeaders::get().CacheControl, "no-cache");
-
-    // Decoding the request should find a cached response that requires validation.
-    // As far as decoding the request is concerned, this is the same as a cache miss with the
-    // exception of injecting validation precondition headers.
-    testDecodeRequestMiss(filter);
-
-    // Make sure validation conditional headers are added
-    const Http::TestRequestHeaderMapImpl injected_headers = {
-        {"if-none-match", etag}, {"if-modified-since", last_modified_date}};
-    EXPECT_THAT(request_headers_, IsSupersetOfHeaders(injected_headers));
-
-    // Encode 304 response
-    // Advance time to make sure the cached date is updated with the 304 date
-    const std::string not_modified_date = formatter_.now(time_source_);
-    Http::TestResponseHeaderMapImpl not_modified_response_headers = {{":status", "304"},
-                                                                     {"date", not_modified_date}};
-
-    // The filter should stop encoding iteration when encodeHeaders is called as a cached response
-    // is being fetched and added to the encoding stream. StopIteration does not stop encodeData of
-    // the same filter from being called
-    EXPECT_EQ(filter->encodeHeaders(not_modified_response_headers, true),
-              Http::FilterHeadersStatus::StopIteration);
-
-    // Check for the cached response headers with updated date
-    Http::TestResponseHeaderMapImpl updated_response_headers = response_headers_;
-    updated_response_headers.setDate(not_modified_date);
-    EXPECT_THAT(not_modified_response_headers, IsSupersetOfHeaders(updated_response_headers));
-
-    // A 304 response should not have a body, so encodeData should not be called
-    // However, if a body is present by mistake, encodeData should stop iteration until
-    // encoding the cached response is done
-    Buffer::OwnedImpl not_modified_body;
-    EXPECT_EQ(filter->encodeData(not_modified_body, true),
-              Http::FilterDataStatus::StopIterationAndBuffer);
-
-    // The filter should add the cached response body to encoded data.
-    Buffer::OwnedImpl buffer(body);
-    EXPECT_CALL(
-        encoder_callbacks_,
-        addEncodedData(testing::Property(&Buffer::Instance::toString, testing::Eq(body)), true));
-
-    // The cache getBody callback should be posted to the dispatcher.
-    // Run events on the dispatcher so that the callback is invoked.
-    dispatcher_->run(Event::Dispatcher::RunType::Block);
-
-    ::testing::Mock::VerifyAndClearExpectations(&encoder_callbacks_);
-
-    filter->onDestroy();
-  }
+  // Add Etag & Last-Modified headers to the response for validation
+  response_headers_.setReferenceKey(Envoy::Http::CustomHeaders::get().Etag, etag);
```

Comment:
optional: this all might be easier to read (and slightly faster) if you put Envoy::Http::CustomHeaders::get() into a temp reference.

Reply:
Done.